QModManager/Injector.cs

```diff
@@ -93,27 +91,35 @@ internal void Remove()
                     Environment.Exit(0);
                 }
 
-                if (File.Exists(backupFilename))
-                {
-                    File.Delete(mainFilename);
+                // Remove backup file if it exists
+                string backupFilePath = Path.Combine(managedDirectory, "Assembly-CSharp.qoriginal.dll");
+                if (File.Exists(backupFilePath))
+                    File.Delete(backupFilePath);
 
-                    File.Move(backupFilename, mainFilename);
+                AssemblyDefinition game = AssemblyDefinition.ReadAssembly(mainFilename);
 
-                    Console.WriteLine();
-                    Console.WriteLine("QModManager was uninstalled successfully");
-                    Console.WriteLine();
-                    Console.WriteLine("Press any key to exit...");
-                    Console.ReadKey();
-                    Environment.Exit(0);
+                TypeDefinition gameInputDef = game.MainModule.GetType("GameInput");
+                MethodDefinition awakeMethod = gameInputDef.Methods.First(x => x.Name == "Awake");
+
+                Instruction patchMethodCall = null;
+
+                foreach(Instruction instruction in awakeMethod.Body.Instructions)
+                {
+                    if(instruction.OpCode == OpCodes.Call && instruction.Operand.ToString().Equals("System.Void QModInstaller.QModPatcher::Patch()"))
+                    {
+                        patchMethodCall = instruction;
+                    }
                 }
 
-                Console.WriteLine();
-                Console.WriteLine("Cannot uninstall, file 'Assembly-CSharp-qoriginal.dll' is missing");
-                Console.WriteLine("To uninstall, you will need to verify game contents ons steam");
+                awakeMethod.Body.GetILProcessor().Remove(patchMethodCall);
```

Comment:
Maybe a `null` check for `patchMethodCall`?

Reply:
Oh yeah, forgot to check for that.